### DISCUSSION TITLE
Insert cell in a specific face

### DISCUSSION THREAD

**Discussion by Qvadi** [iqm-finland/KQCircuits#74] [Q&A] [answered]
Hi,
Is it possible to add a waveguide coplanar element in a specific face id?

for example, 
`self.insert_cell(WaveguideCoplanar, face_id = "2b1")`

It seems like `insert_cell()` only inserts elements in the default face_id which is set by the face_ids[0] list. Can you on-demand do this somehow?

**Comment by qpavsmi** [comment, marked as answer]
Hi,
the face where the geometry of the cell are placed at are controlled by the `face_ids` parameter. The parameter takes a list of face strings as value, by default set to `["1t1", "2b1", "1b1", "2t1"]`. In the element class code we point to the index in that list to place appropriate geometry at appropriate faces. For example a flipchip will point to 0 in `face_ids` => "1t1" to define geometry of the bottom chip and point to 1 in `face_ids` => "2b1" to define geometry on the top chip. But if we would like to invert this order, passing a `"face_ids": ["2b1", "1t1"]` parameter value would do that.

So, to fix your issue, since `insert_cell` can also take parameters for the element class you are adding, then `self.insert_cell(WaveguideCoplanar, {"face_ids": ["2b1", "1t1"]})` should work

**Reply by Qvadi** [reply]
Thanks! It works. To restate the problem I was trying to solve: 
I had two `insert_cell` instances, one which is built out of `waveguide_composite` and the other one with `meander`. For the `waveguide_composite`, layer changing works well. But when you insert the `meander` at the position where there `waveguide_composite` ends, the layer continuity was not maintained inspite of me passing the argument `face_id="2b1"` like you do when you want to change layers in `waveguide_composite` definition. This didnt work.

Solution: what worked as suggested by qpavsmi is a slightly modified version `insert_cell(Meander, face_ids=["1t1", "2b1"])` along with other parameters for the meander element.
